apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/PreAdjFissClaimResponseTransformerV2.java

```diff
@@ -0,0 +1,150 @@
+
+  static {
+    STATUS_TO_OUTCOME = new HashMap<>();
+    STATUS_TO_OUTCOME.put(' ', ClaimResponse.RemittanceOutcome.QUEUED);
+    STATUS_TO_OUTCOME.put('a', ClaimResponse.RemittanceOutcome.QUEUED);
+    STATUS_TO_OUTCOME.put('s', ClaimResponse.RemittanceOutcome.PARTIAL);
+    STATUS_TO_OUTCOME.put('p', ClaimResponse.RemittanceOutcome.COMPLETE);
+    STATUS_TO_OUTCOME.put('d', ClaimResponse.RemittanceOutcome.ERROR);
+    STATUS_TO_OUTCOME.put('i', ClaimResponse.RemittanceOutcome.PARTIAL);
+    STATUS_TO_OUTCOME.put('r', ClaimResponse.RemittanceOutcome.ERROR);
+    STATUS_TO_OUTCOME.put('t', ClaimResponse.RemittanceOutcome.ERROR);
+    STATUS_TO_OUTCOME.put('m', ClaimResponse.RemittanceOutcome.PARTIAL);
+  }
+
+  /**
+   * @param metricRegistry the {@link MetricRegistry} to use
+   * @param claimEntity the FISS {@link PreAdjFissClaim} to transform
+   * @return a FHIR {@link ClaimResponse} resource that represents the specified claim
+   */
+  @Trace
+  static ClaimResponse transform(MetricRegistry metricRegistry, Object claimEntity) {
+    Timer.Context timer = metricRegistry.timer(METRIC_NAME).time();
+
+    if (!(claimEntity instanceof PreAdjFissClaim)) {
+      throw new BadCodeMonkeyException();
+    }
+
+    ClaimResponse claim = transformClaim((PreAdjFissClaim) claimEntity);
+
+    timer.stop();
+    return claim;
+  }
+
+  /**
+   * @param claimGroup the {@link PreAdjFissClaim} to transform
+   * @return a FHIR {@link ClaimResponse} resource that represents the specified {@link
+   *     PreAdjFissClaim}
+   */
+  private static ClaimResponse transformClaim(PreAdjFissClaim claimGroup) {
+    ClaimResponse claim = new ClaimResponse();
+
+    claim.setId("f-" + claimGroup.getDcn());
+    claim.setMeta(new Meta().setLastUpdated(Date.from(claimGroup.getLastUpdated())));
+    claim.setExtension(getExtension(claimGroup));
+    claim.setIdentifier(getIdentifier(claimGroup));
+    claim.setStatus(ClaimResponse.ClaimResponseStatus.ACTIVE);
+    claim.setOutcome(STATUS_TO_OUTCOME.get(Character.toLowerCase(claimGroup.getCurrStatus())));
+    claim.setType(getType());
+    claim.setUse(ClaimResponse.Use.CLAIM);
+    claim.setCreated(new Date());
+    claim.setInsurer(new Reference().setIdentifier(new Identifier().setValue("CMS")));
+    claim.setPatient(getPatient(claimGroup));
+    claim.setRequest(new Reference(String.format("Claim/f-%s", claimGroup.getDcn())));
+
+    return claim;
+  }
+
+  private static List<Extension> getExtension(PreAdjFissClaim claimGroup) {
+    return Collections.singletonList(
+        new Extension()
+            .setUrl("https://dcgeo.cms.gov/resources/variables/fiss-status")
+            .setValue(
+                new Coding(
+                    "https://dcgeo.cms.gov/resources/variables/fiss-status",
+                    "" + claimGroup.getCurrStatus(),
+                    STATUS_TEXT.get(Character.toLowerCase(claimGroup.getCurrStatus())))));
```

Comment:
Just curious: what happens if the `claimGroup.getCurrStatus()` isn't a valid key so `get` returns null?

Reply:
This is a good question, we'll need to define does it return a default value, or just empty.  Also, I don't quite remember, was there a question of if the current status could be null?  I know our schema currently has it as `not null`, but if it could, that would be another issue as well to address.

Reply:
Good point about the claim status being non-null.